src/foremast/utils/dns.py

```diff
@@ -148,6 +149,13 @@ def update_failover_dns_record(env, zone_id, **kwargs):
     zone_name = hosted_zone_info['HostedZone']['Name'].rstrip('.')
     dns_name = kwargs.get('dns_name')
 
+    #Check that the primary record exists
+    failover_state = kwargs.get('failover_state')
+    if failover_state.lower() != 'primary':
```

Comment:
Actually, nevermind, the `primary_record_exists()` is an necessary, if `failover_state` isn't primary. Its also makes an extra api call which could be expensive and unwarranted in this scenario.